gwt-jackson/src/main/java/com/github/nmorel/gwtjackson/client/ser/BaseDateJsonSerializer.java

```diff
@@ -53,7 +55,7 @@ private DateJsonSerializer() { }
 
         @Override
         protected void doSerialize( JsonWriter writer, @Nonnull Date value, JsonSerializationContext ctx, JsonSerializerParameters params ) {
-            if ( ctx.isWriteDatesAsTimestamps() || params.getShape().isNumeric() ) {
+            if ( (ctx.isWriteDatesAsTimestamps() || params.getShape().isNumeric()) && params.getShape() != Shape.STRING ) {
```

Comment:
I had to change this to mimic jackson behaviour that @JsonFormat has priority over global WriteDatesAsTimestamps feature